src/main.ts

```diff
@@ -135,7 +142,25 @@ export function generateFile(typeMap: TypeMap, fileDesc: FileDescriptorProto, pa
   }
 
   visitServices(fileDesc, sourceInfo, (serviceDesc, sInfo) => {
-    file = file.addInterface(generateService(typeMap, fileDesc, sInfo, serviceDesc, options));
+    file = file.addInterface(
+      options.nestJs
+        ? generateNestjsService(typeMap, fileDesc, sInfo, serviceDesc, options)
+        : generateService(typeMap, fileDesc, sInfo, serviceDesc, options)
+    );
+    if (options.nestJs) {
+      // generate nestjs grpc client interface
+      file = file.addInterface(generateNestjsServiceClient(typeMap, fileDesc, sInfo, serviceDesc, options));
+
+      // generate nestjs grpc service controller decorator
+      file = file.addFunction(generateNestjsGrpcServiceMethodsDecorator(serviceDesc, options));
+
+      let serviceConstName =  `${camelToSnake(serviceDesc.name)}_NAME`;
+      if(!serviceDesc.name.toLowerCase().endsWith('service')){
+        serviceConstName = `${camelToSnake(serviceDesc.name)}_SERVICE_NAME`;
+      }
+
+      file = file.addCode(CodeBlock.empty().add(`export const %L = '%L'`, serviceConstName, serviceDesc.name));
```

Comment:
Minor but throw a `;` after the `'%'`?

Reply:
But the `add` function automatically adds this ending `;` right? Cause the code that it formats now isn't missing a `;`, and when looking at other parts of the code you're doing it the same. Or I don't get it hehe ;)

Reply:
Hm, I'm pretty sure `add` doesn't suffix a `;` enough though some of these methods like `beginControlFlow`/`endControlFlow` do.

And the output doesn't have them now:

```
export const HERO_PACKAGE_NAME = 'hero'
export const HERO_SERVICE_NAME = 'HeroService'
```

Reply:
@stephenh woow you're totally right! My bad! ;) will fix this